### DISCUSSION TITLE
Add Readeck / Authentication

### DISCUSSION THREAD

**Discussion by MrColumbo** [Waboodoo/HTTP-Shortcuts#406] [Q&A]
Dear All,

i just learned about this app and now I am trying to figure out how to use to act as a share to app and basically send a URL to my readeck instance ( https://readeck.org/en/). While the author from readeck might look into that himself and probably knows way better how to do that i thought that this app is meant to get used with different services without the need to get their support (I guess linkedin did not support that app but it looks like there is a way to use this app with liinkedin).

Well - i looked at how readeck uses a post request to send the url and that portion should be not to complicated (hopefully :)  ......but where i am not even sure on how to start is the authentication piece.

I know that readeck does not use HTTP basic authentiation - that would be to easy. When I look at a post request i see some cas like csrf_key and maybe i just need to use a static var. 

I also thought that authentication is done in a way throught the web app that once I logged in it will use a session and all the information i see in a POST request after i am logged in will not be helpful since the session id will expire and maybe HTTP Shortcuts does not work with all web services if they are not using some authentication method which allows third party apps like this to integrate. In that case I am wondering if there is an easy way to identify that. 

Well - are there any special key words i need to look in HTTP requests to see if there is a way to integrate?

thanks a lot for any feedback :)

**Comment by Waboodoo** [comment]
Hi,
I had a quick glance at the Readeck repository and website but couldn't find any documentation on how it does authentication. I recommend reaching out to the author(s) of Readeck.

From the side of HTTP Shortcuts, there are 4 types of authentication that are built-in, all of which can be found in a shortcut's Authentication section, but my guess is that none of them will work for Readeck. There's likely some mechanism in Readeck that let's you obtain an auth token, as well as a CSRF token which is also required. Both of these are likely stored in cookies. They might be short-lived, so possibly you'll need to add a dedicated shortcut which calls a sign-in endpoint to renew these cookies every now and then. If they are not stored in cookies, but maybe a header, then you might have to store these values yourself by putting them into a variable and then using that variable in a header. For this I can just point you at the documentation for [Scripting](https://http-shortcuts.rmy.ch/scripting) and [Variables](https://http-shortcuts.rmy.ch/variables). There you'll find examples of how to read data from headers or other parts of the response and store it into variables.
Hope this helps, good luck.
